apps/filebrowser/src/filebrowser/views.py

```diff
@@ -1190,6 +1190,11 @@ def generic_op(form_class, request, op, parameter_names, piggyback=None, templat
 
       if next:
         logging.debug("Next: %s" % next)
+        file_path_prefix = '/filebrowser/view='
+        if next.startswith(file_path_prefix):          
+          decoded_file_path = next[len(file_path_prefix):]
+          filepath_encoded_next = file_path_prefix + urllib_quote(decoded_file_path.encode('utf-8'), safe=SAFE_CHARACTERS_URI_COMPONENTS)
+          return format_preserving_redirect(request, filepath_encoded_next)
         # Doesn't need to be quoted: quoting is done by HttpResponseRedirect.
         return format_preserving_redirect(request, next)
```

Comment:
Suggestion: Do we need 2 return statements here?

```python
next = filepath_encoded_next
```
and the last return can handle it?

Reply:
I don't know if there could be a case where the 

> Was this tested with Py3 Hue also?

Yes

Reply:
Ahh sorry, now I understand what you mean. True, I could have overwritten "next" instead.